src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -1215,11 +1215,15 @@ await this.CommitOutboundQueueMessages(
                     this.orchestrationSessionManager.AddMessageToPendingOrchestration(session.ControlQueue, messages, session.TraceActivityId, CancellationToken.None);
                 }
             }
-            catch (RequestFailedException rfe) when (rfe.Status == (int)HttpStatusCode.PreconditionFailed)
+            // If this is the first work item for the orchestration and it has already been completed, then session.Etag will be null and a new history will attempt to be inserted when one already exists.
```

Comment:
nits - feel free to disregard.
```suggestion
            // If this is the first work item for the orchestration but it was already completed, by another worker then our session.Etag will be null. This means we'll try to create a new history despite one already existing.

```